Apromore-Custom-Plugins/Process-Simulation-Info-Logic/src/main/java/org/apromore/processsimulation/service/SimulationInfoService.java

```diff
@@ -84,11 +91,16 @@ public class SimulationInfoService {
 
     private final SimulationInfoConfig config;
     private final CalendarService calendarService;
+    private UserMetadataService userMetadataService;
+    private ObjectMapper objectMapper;
 
     @Autowired
-    public SimulationInfoService(SimulationInfoConfig config, CalendarService calendarService) {
+    public SimulationInfoService(SimulationInfoConfig config, CalendarService calendarService,
+                                 UserMetadataService userMetadataService, ObjectMapper objectMapper) {
```

Comment:
I'm wondering if passing the object mapper in as a bean to be auto wired is a good idea. This would mean that if someone else is making use of the same object mapper, they could potentially change its properties, which in turn may affect the deserialisation behaviour here.

Would it be safer to create an instance of ObjectMapper within this constructor instead.
i.e. in line 103
this.objectMapper = new ObjectMapper();